janitor/io.py

```diff
@@ -63,9 +65,68 @@ def read_csvs(
                     "Files cannot be concatenated"
                 )
         return pd.concat(
-            list(dfs_dict.values()),  # noqa: PD011
+            list(dfs_dict.values()),
             ignore_index=True,
-            sort=False,
+            sort=False,  # noqa: PD011
         )
     else:
         return dfs_dict
+
+
+def read_commandline(cmd: str, **kwargs) -> pd.DataFrame:
+    """
+    Read a CSV file based on a command-line command.
+
+    For example, you may wish to run the following command on `sep-quarter.csv`
+    before reading it into a pandas DataFrame:
+
+    ```bash
+    cat sep-quarter.csv | grep .SEA1AA
+    ```
+
+    In this case, you can use the following Python code to load the dataframe:
+
+    ```python
+    import janitor as jn
+    df = jn.io.read_commandline("cat data.csv | grep .SEA1AA")
+
+    This function assumes that your command line command will return
+    an output that is parsable using pandas.read_csv and StringIO.
+    We default to using pd.read_csv underneath the hood.
+    Keyword arguments are passed through to read_csv.
+    ```
+
+    :param cmd: Shell command to preprocess a file on disk.
+    :param kwargs: Keyword arguments that are passed through to pd.read_csv().
+    :returns: A pandas DataFrame parsed from the stdout of the underlying
+        shell.
+    :raises TypeError: If the input is anything except a string.
+    :raises EmptyDataError: If there is no data to parse, this often happens
+        because the cmd param is an invalid bash command,
+        thus nothing happens in the shell.
+    """
+    # cmd = cmd.split(" ")
+
```

Comment:
the subprocess will return an error if the return code is non zero, since capture_output is set to True. so maybe we should use that instead (agreeing with @thatlittleboy ):

```py
from .utils import check # at the top
check('cmd', cmd, str)
outcome = subprocess.run(
            cmd, shell=True, capture_output=True, text=True
        )
```
removes the need for the try clause for the shell section.